telegram/inputfile.py

```diff
@@ -69,19 +70,30 @@ def __init__(self,
             self.input_name = 'certificate'
             self.input_file = data.pop('certificate')
 
-        if isinstance(self.input_file, file):
+        if str(self.input_file).startswith('http'):
+            from_url = True
+            self.input_file = urlopen(self.input_file)
+        else:
+            from_url = False
+
+        if isinstance(self.input_file, file) or from_url:
             self.input_file_content = self.input_file.read()
             if 'filename' in data:
                 self.filename = self.data.pop('filename')
-            else:
+            elif isinstance(self.input_file, file):
                 self.filename = os.path.basename(self.input_file.name)
-            self.mimetype = mimetypes.guess_type(self.filename)[0] or \
-                DEFAULT_MIME_TYPE
-
-        if 'http' in self.input_file:
-            self.input_file_content = urlopen(self.input_file).read()
-            self.mimetype = InputFile.is_image(self.input_file_content)
-            self.filename = self.mimetype.replace('/', '.')
+            elif from_url:
+                self.filename = \
+                    self.input_file.url.split('/')[-1]\
+                        .split('?')[0].split('&')[0]
```

Comment:
These splits assume the last argument of a GET query will be a file name, right?